### PR TITLE
feat: Add admin script to change Finder address

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

This script will need to be used to upgrade the "Oracle" implementation address on the Polygon Finder so that we can upgrade the `OracleChildTunnel`. See #3863 for motivation for upgrading `OracleChildTunnel`. I tested this locally on mainnet fork.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [X]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/scripts/src/admin-proposals/changeFinderAddress.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      try {
+        const implementationAddress = await mainnetContracts.finder.methods
+          .getImplementationAddress(utf8ToHex(contract))
+          .call();
+        if (implementationAddress === toChecksumAddress(ethereum)) {
+          skip = true;
+          console.log("- Contract @ ", ethereum, `is already set to ${contract} in the Finder. Nothing to do.`);
+        }
+      } catch (err) {
+        // eslint-disable-line prettier/prettier
```
Comment (mrice32) [inline packages/scripts/src/admin-proposals/changeFinderAddress.js:105]: Why do we ignore errors?

File: packages/scripts/src/admin-proposals/changeFinderAddress.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      try {
+        const implementationAddress = await contractsByNetId[137].finder.methods
+          .getImplementationAddress(utf8ToHex(contract))
+          .call();
+        if (implementationAddress === toChecksumAddress(polygon)) {
+          skip = true;
+          console.log("- Contract @ ", polygon, `is already set to ${contract} in the Finder. Nothing to do.`);
+        }
+      } catch (err) {
+        // eslint-disable-line prettier/prettier
```
Comment (mrice32) [inline packages/scripts/src/admin-proposals/changeFinderAddress.js:129]: Same question here?

File: packages/scripts/src/admin-proposals/changeFinderAddress.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        try {
+          const implementationAddress = await contractsByNetId[network.chainId].finder.methods
+            .getImplementationAddress(utf8ToHex(contract))
+            .call();
+          if (implementationAddress === toChecksumAddress(network.value)) {
+            skip = true;
+            console.log("- Contract @ ", network.value, `is already set to ${contract} in the Finder. Nothing to do.`);
+          }
+        } catch (err) {
+          // eslint-disable-line prettier/prettier
```
Comment (mrice32) [inline packages/scripts/src/admin-proposals/changeFinderAddress.js:160]: Same question

**Review by mrice32 — APPROVED** [review]

File: packages/scripts/src/admin-proposals/changeFinderAddress.js (hunk 1 above)
Comment (nicholaspai) [reply packages/scripts/src/admin-proposals/changeFinderAddress.js:105]: Added a comment:

```
        // If `getImplementationAddress` reverts, then implementation is not set yet so we shouldn't skip this
        // proposal. Not having an implementation set is an expected situation for this script so we should ignore this
        // error.
```

File: packages/scripts/src/admin-proposals/changeFinderAddress.js (hunk 2 above)
Comment (nicholaspai) [reply packages/scripts/src/admin-proposals/changeFinderAddress.js]: answered above

File: packages/scripts/src/admin-proposals/changeFinderAddress.js (hunk 3 above)
Comment (nicholaspai) [reply packages/scripts/src/admin-proposals/changeFinderAddress.js]: answered above

**Pull request merged**